python/cairo-core/src/cairo_core/maths.cairo

```diff
@@ -536,3 +536,57 @@ func felt252_bit_length{range_check_ptr}(value: felt) -> felt {
 
     return bit_length;
 }
+
+func felt252_to_bits{range_check_ptr}(limb: felt, bits_ptr: felt*, current_len: felt) -> felt {
+    alloc_locals;
+    if (limb == 0) {
+        return current_len;
+    }
+    with_attr error_message("felt252_to_bits: limb must be < 252") {
+        assert [range_check_ptr] = limb;
+        assert [range_check_ptr + 1] = 251 - limb;
+        let range_check_ptr = range_check_ptr + 2;
+    }
+
+    let output = &bits_ptr[0];
+    %{ felt252_to_bits %}
+
+    tempvar range_check_ptr = range_check_ptr;
+    tempvar current_len = current_len;
+    tempvar i = 0;
+    tempvar acc = 0;
+
+    loop:
+    let range_check_ptr = [ap - 3];
+    let current_len = [ap - 2];
+    let limb = [ap - 1];
+
+    let is_done = is_zero(limb);
+    jmp end if is_done != 0;
+
+    // Get the current bit using modulo 2
+    let (_, bit) = unsigned_div_rem(limb, 2);
+    assert bits_ptr[current_len] = bit;
+
+    // all the generated felt elements should be either 0 or 1
+    let bit_minus_one = bit - 1;
+    let product = bit * bit_minus_one;
+    let is_valid_bit = is_zero(product);
```

Comment:
// # 1. Validate that all intermediate bits are 0 or 1
this can be simply translated into `assert bit * bit = bit`